src/deepsparse/loggers/function_logger.py

```diff
@@ -31,49 +33,76 @@ class FunctionLogger(BaseLogger):
     according to FunctionLogger's attributes
 
     :param logger: A child DeepSparse Logger object
-    :param target_identifier: The string that needs to match the
-        `identifier` (argument to the log() method),
-        so that the FunctionLogger applies
-        the metric function and logs the data
-    :param function_name: Name of the metric function
-    :param function: The metric function to be applied
-    :param frequency: The frequency with which the metric
-        name is to be applied
+    :param target_to_metric_function_configs: A mapping from target identifier to
+        a list of MetricFunctionConfig objects
     """
 
     def __init__(
         self,
         logger: BaseLogger,
-        target_identifier: str,
-        function: Callable[[Any], Any],
-        function_name: str = None,
-        frequency: int = 1,
+        target_to_metric_function_configs: Dict[
```

Comment:
Bumping this: why are we changing FunctionLogger in this PR? Why can't you use the existing FunctionLogger?

Reply:
this was discussed in the function logger PR, the goal is to remove the need for an extra multi logger that wraps all of the function loggers, additionally, match and extract only need to be run once under this configuration. note that the old functionality is still in tact if given a single function

Reply:
Yeah I"m good with that - which is exactly what the current FunctionLogger does (and what was merged in on Friday). I'm asking why are there any changes to FunctionLogger in this PR? Why the switch to passing a config object instead of what is there now?